File: PGS/Lv.2/타겟 넘버.cpp
#include <bits/stdc++.h>
using namespace std;

int answer = 0;

void dfs(vector<int> &numbers, int target, int start, int num, int depth) {
	if (depth == numbers.size()) {
		if (num == target) {
			answer++;
		}
		return;
	}
	dfs(numbers, target, start + 1, num + numbers[start], depth + 1);
	dfs(numbers, target, start + 1, num - numbers[start], depth + 1);
}

int solution(vector<int> numbers, int target) {

	dfs(numbers, target, 0, 0, 0);
	return answer;
}

File: PGS/Lv.2/JAVA/타겟 넘버.java
class Solution {
    
    static int n, answer;
    
    void dfs(int[] numbers, int target, int num, int depth) {
        if(depth == n) {
            if(num == target) answer++;
            return;
        }
        
        dfs(numbers, target, num + numbers[depth], depth + 1);
        dfs(numbers, target, num - numbers[depth], depth + 1);
    }
    
    public int solution(int[] numbers, int target) {
        answer = 0;
        n = numbers.length;
        
        dfs(numbers, target, 0, 0);
        
        return answer;
    }
}